Tools/YamlCreate.ps1

```diff
@@ -1,262 +1,261 @@
+} while ($Tags.Length -gt 40)
 
 do {
-    $Homepage = Read-Host -Prompt   '[OPTIONAL] Enter the Url to the homepage of the application'
-} while ($Homepage.length -ge 1 -AND ($Homepage.Length -lt 10 -or $Homepage.Length -gt 2000))
+    $Homepage = Read-Host -Prompt '[OPTIONAL] Enter the Url to the homepage of the application' | TrimString
+} while (-not [string]::IsNullOrWhiteSpace($LicenseUrl) -and ($Homepage.Length -lt 10 -or $Homepage.Length -gt 2000))
 
 do {
-    $Description = Read-Host -Prompt '[OPTIONAL] Enter a description of the application'
-} while ($Description.length -gt 500)
+    $Description = Read-Host -Prompt '[OPTIONAL] Enter a description of the application' | TrimString
+} while ($Description.Length -gt 500)
 
 # Only prompt for silent switches if $InstallerType is "exe"
-if ($InstallerType.ToLower() -eq "exe") {
-    $Silent = Read-Host -Prompt '[OPTIONAL] Enter the silent install switch'
-    $SilentWithProgress = Read-Host -Prompt '[OPTIONAL] Enter the silent (with progress) install switch'
+if ($InstallerType -ieq "exe") {
+    $Silent = Read-Host -Prompt '[OPTIONAL] Enter the silent install switch'| TrimString
+    $SilentWithProgress = Read-Host -Prompt '[OPTIONAL] Enter the silent (with progress) install switch'| TrimString
 }
 
-
-##########################################
-# Write  metadata
+#endregion
 ##########################################
 
-$OFS
-$string = "Id: " + $id
-write-output $string | out-file $filename
-write-host "Id: "  -ForeGroundColor Blue -NoNewLine
-write-host $id  -ForeGroundColor White
-
-$string = "Version: " + $Version
-write-output $string | out-file $filename -append
-write-host "Version: "  -ForeGroundColor Blue -NoNewLine
-write-host $Version -ForeGroundColor White
-
-
-$string = "Name: " + $AppName
-write-output $string | out-file $filename -append
-write-host "Name: "  -ForeGroundColor Blue -NoNewLine
-write-host $AppName  -ForeGroundColor White
-
-$string = "Publisher: " + $Publisher
-write-output $string | out-file $filename -append
-write-host "Publisher: "  -ForeGroundColor Blue -NoNewLine
-write-host $Publisher -ForeGroundColor White
-
-$string = "License: " + $License
-write-output $string | out-file $filename -append
-write-host "License: "  -ForeGroundColor Blue -NoNewLine
-write-host $License  -ForeGroundColor White
-
-if (!($LicenseUrl.length -eq 0)) {
-
-    $string = "LicenseUrl: " + $LicenseUrl
-    write-output $string | out-file $filename -append
-    write-host "LicenseUrl: "  -ForeGroundColor Blue -NoNewLine
-    write-host $LicenseUrl  -ForeGroundColor White
-
+##########################################
+#region Write metadata
+
+# YAML files should always start with the document start separator "---"
+# https://yaml.org/spec/1.2/spec.html#id2760395
+$string = "---$NewLine"
+Write-Output $string | Out-File $ManifestName
+
+Write-Host $NewLine
+$string = "Id: $ID"
```

Comment:
this adds another newline after the "---" yaml header. I'm not a yaml expert but the examples on [https://yaml.org/spec/1.2/spec.html#id2760395](https://yaml.org/spec/1.2/spec.html#id2760395), the text follow --- directly follows.  Is that expected?

Here's the output when running your PR:

---
       <----- there's a newline here.
Id: skylot.jadx
Version: 1.1.0
Name: jadx
Publisher: skylot

Reply:
The new line is optional but makes it more pleasing to the human eye. In its place there could be 0-n new lines.

Reply:
Makes sense to me.

---
In reply to: [476890201](https://github.com/microsoft/winget-pkgs/pull/3241#discussion_r476890201) [](ancestors = 476890201)